src/prepare_host_handler.cpp

```diff
@@ -67,8 +67,6 @@ void PrepareHostHandler::prepare(uv_loop_t* loop, const ConnectionSettings& sett
 }
 
 void PrepareHostHandler::on_close(Connection* connection) {
-  callback_(this);
-
```

Comment:
@mpenick What do you think of this idea?  The callback here is Cluster::on_prepare_host_up and it doesn't look like there's any benefit to calling it when we're shutting down the cluster anyway.  And if we don't call it we should avoid tangling up Cluster and PrepareHostHandler method calls... which should remove the possibility of a pure virtual function call.

Does this make sense or am I missing something?

Reply:
I think this is going after the symptom instead of fixing the cause. My bet is the listener which is `SessionBase` is being freed before the the Cluster object is finished with it.